AstSemantics.md

```diff
@@ -248,6 +295,32 @@ The details of index space for local variables and their types will be further c
 e.g. whether locals with type `i32` and `i64` must be contiguous and separate from
 others, etc.
 
+## Global variables
+
+A *global variable* stores a single value of a fixed [value type](#types) and may be
+declared either *mutable* or *immutable*. This provides WebAssembly with memory
+locations that are disjoint from any [linear memory](#linear-memory) and thus
+cannot be arbitrarily aliased as bits.
+
```

Comment:
For these global variables to be dynamically importable, or 'instantiation-time immutable values', it would appear that they can not be baked into the code in AOT compilation. If so then it is not clear what utility they have, and they would not appear to really be necessary?

I see some potential for the compiler to optimize code knowing that they will not change, similar to a `const var`, but that does not appear to be a core part of this PR? Also `let` variables have this property too, so if these were added then these might help address some of these optimization opportunities.

I recall mention of patchable values in prior discussion, and are these global constant variables intended to meet this use case? This would appear to require each instance to have a separate copy of the code, but that would somewhat defeat the purpose of sharing code.

For immutable global variables defined in the module, could they be baked into the code AOT?

There might be some significant performance burdens for shared code, extra dereferences. Does the wasm model allow the web developer to make the choice on these trade offs. Should a module have a flag to indicate that it is likely to be shared code, so that only in this case the runtime might avoid re-compiling or copying the code for each instance.